2005-09-17 17:52  Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
	- vdr.c: use daemon(3) instead of forking manually; this
	  executes 'setsid()' and creates better FDs 0-2

--- vdr-1.3.32/vdr.c.daemon	2005-09-17 13:40:57.000000000 +0200
+++ vdr-1.3.32/vdr.c	2005-09-17 13:46:41.000000000 +0200
@@ -508,17 +508,11 @@
   // Daemon mode:
 
   if (DaemonMode) {
-     pid_t pid = fork();
-     if (pid < 0) {
-        fprintf(stderr, "%m\n");
+     if (daemon(1, 0)==-1) {
         esyslog("ERROR: %m");
+        perror("daemon()");
         return 2;
-        }
-     if (pid != 0)
-        return 0; // initial program immediately returns
-     fclose(stdin);
-     fclose(stdout);
-     fclose(stderr);
+       }
      }
   else if (Terminal) {
      // Claim new controlling terminal
